runtime/compiler/rpc/J9Client.hpp

```diff
@@ -82,11 +95,58 @@ class J9ClientStream : J9Stream
 
    void shutdown();
 
+   template <typename ...T>
+   void writeError(J9ServerMessageType type, T... args)
+      {
+      _cMsg.set_type(type);
+      if (type == J9ServerMessageType::compilationAbort)
+         _cMsg.mutable_data()->clear_data();
+      else
+         setArgs<T...>(_cMsg.mutable_data(), args...);
```

Comment:
Question: for clientTerminate we only send clientID. Are all the other fields clear or do we have to clear them first and then do `setArgs`?

Reply:
`setArgs` will overwrite things that were previously in `_cMsg.mutable_data()`, there is no need to first clear them.

Reply:
`mutable_data()` will be re-constructed based on the `T... args` we pass in.